src/SignUp.tsx

```diff
@@ -0,0 +1,152 @@
+// import axios from 'axios';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router';
+
+const SignUp = () => {
+	const navigate = useNavigate();
+	const [errors, setErrors] = useState('');
+	const [values, setValues] = useState({
+		firstName: '',
+		lastName: '',
+		email: '',
+		password: '',
+		confirmPassword: '',
+	});
+
+	const handleChange = (e: { target: { name: any; value: any } }) => {
+		const { name, value } = e.target;
+		setValues({
+			...values,
+			[name]: value,
+		});
+	};
+
+	const handleValidation = () => {
+		if (
+			values.firstName === '' ||
+			values.lastName === '' ||
+			values.email === '' ||
+			values.password === '' ||
+			values.confirmPassword === ''
+		) {
+			setErrors('some field are incomplete');
+		} else if (
+			values.confirmPassword.length < 3 &&
+			values.password.length < 3
+		) {
+			setErrors('password must be more then 3 characters');
+		} else if (values.password !== values.confirmPassword) {
+			setErrors('Password and Confirm Password does NOT match');
+		} else if (values.email.includes('@') === false) {
+			setErrors('Email invalid');
+		} else {
+			// axios
+			// .post('http://localhost:8080/customers', values)
+			// .then(response => console.log(response, 'submitted'))
+			// .catch(error => {
+			// 	if (error.toJSON().message === 'Network Error') {
+			// 		alert('error');
+			// 	}
+			// });
+			navigate('/login');
+		}
+	};
+
+	const handleSubmit = (e: any) => {
+		e.preventDefault();
+		handleValidation();
+	};
+
+	return (
+		<div>
+			<h1> This is the SignUp Page </h1>
+			<form className="form">
+				<div className="form-body">
+					<div>
+						<label htmlFor="firstname">
+							First Name
```

Comment:
First Name, Last Name, Email, Password, Confirm Password, and Submit; did you intend to leave them unwrapped?

Reply:
They are wrapped around in  `label`

Reply:
I see, got it!